src/js/renderers.js

```diff
@@ -0,0 +1,49 @@
+var utils = require('./utils');
+
+var gene = function(config) {
+  return function(selection) {
+    var row_elements = selection.selectAll('g').data(function(d) { return d; })
```

Comment:
Good question. See Bostock's explanaition on nested selections: http://bost.ocks.org/mike/nest/#data. I'll also add this as a comment.